src/library/filescanner.c

```diff
@@ -1816,12 +1816,62 @@ playlist_add_path(FILE *fp, int pl_id, const char *path)
   return 0;
 }
 
+static int
+ensure_file_record(const char *path)
```

Comment:
Is it not better to expect the virtual_path instead of the path? `playlist_add_files` gets the virtual_path and in `queue_save` we also have the virtual_path in the queue_item. It is not always possible to translate a virtual_path to the path (e. g. spotify songs have 'spotify:/{artist}/{album}/{title}' as virtual path while the path is something like 'spotify:track:3n3Ppam7vgaVa1iaRUc9Lp').

Reply:
The virtual path is most probably better. I just don't know enough about the inner design to be able to decide the proper way. So I basically cut and pasted my way to a solution that *just works* :-)